github/resource_github_branch_protection.go

```diff
@@ -336,8 +354,15 @@ func resourceGithubBranchProtectionDelete(d *schema.ResourceData, meta interface
 }
 
 func buildProtectionRequest(d *schema.ResourceData) (*github.ProtectionRequest, error) {
+	requireLinearHistory := d.Get("require_linear_history").(bool)
+	allowForcePushes := d.Get("allow_force_pushes").(bool)
+	allowDeletions := d.Get("allow_deletions").(bool)
+
 	req := &github.ProtectionRequest{
-		EnforceAdmins: d.Get("enforce_admins").(bool),
+		EnforceAdmins:        d.Get("enforce_admins").(bool),
+		RequireLinearHistory: &requireLinearHistory,
+		AllowForcePushes:     &allowForcePushes,
+		AllowDeletions:       &allowDeletions,
```

Comment:
Nit: we can also use the `go-github` library's conversion method `github.Bool()`